src/Impostor.Server/Net/Inner/Objects/InnerPlayerControl.cs

```diff
@@ -688,15 +749,50 @@ private async ValueTask<bool> HandleMurderPlayer(ClientPlayer sender, IInnerPlay
 
             if (target != null && !target.PlayerInfo.IsDead)
             {
-                ((InnerPlayerControl)target).Die(DeathReason.Kill);
-                await _eventManager.CallAsync(new PlayerMurderEvent(Game, sender, this, target));
+                if ((result & (MurderResultFlags.FailedError | MurderResultFlags.FailedProtected)) == 0)
```

Comment:
1. Wouldn't it be easier to check for `Succeeded`?
2. We do this check in 2 places, so we probably should make a `MurderResultFlags#IsFailed` extension method

Reply:
1. No, because it also succeeded if it was decided by the host
2. Not opposed. Handling this bitflag is annoying

Reply:
1. I mean `(result & MurderResultFlags.Succeeded) != 0`